initramfs/init.sh

```diff
@@ -0,0 +1,190 @@
+BB_BIN="/bin/busybox"
+
+init="/sbin/init"
+ro=""
+root=""
+rootflags=""
+rootfstype="auto"
+
+shutdown()
+{
+	while [ 1 ]; do
+		poweroff
+		echo "Please remove power to complete shutdown."
+		sleep 10s
+	done
+}
+
+restart()
+{
+	echo "Rebooting in 5 seconds ..."
+	sleep 5s
+	reboot
+	watchdog -T 1 -t 60 -F /dev/watchdog
+	shutdown
+}
+
+rescue_shell()
+{
+	# TODO: part 4 of EM-2396
+	${BB_BIN} echo -e "\n" \
+		"##################################################\n" \
+		"#                   ▄▄▄      ▄▄▄                 #\n" \
+		"#                   ███████▄▄███▄                #\n" \
+		"#                 ▄███ ▄▄▄   ▄▄▄▐█               #\n" \
+		"#              ▄██  ██ ▀▀▀   ▀▀▀▐█▄              #\n" \
+		"#              ███▄▄██ ▄▄▄▄▄▄▄▄▄▐██              #\n" \
+		"#               ██████          ▐█               #\n" \
+		"#            ▄████▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀█▄           #\n" \
+		"#          ████▐█ ▐████████████████▌▐██▄         #\n" \
+		"#        ▐███▌▄▄█ ▐████████████████▌▐█▄█         #\n" \
+		"#         ███████ ███████▀▀ ▀██████▌▐██▌         #\n" \
+		"#         ███████ █████████ ██▐████▌▐██▌         #\n" \
+		"#        ▄███████ ██████▄██████████▌▐█▀█         #\n" \
+		"#        ████  ▐█ ▐████████████████▌▐█ █▌        #\n" \
+		"#        ▀███ ███   ▀▀▀▀            ▐██▀         #\n" \
+		"#           ▀▀████▄▄████████████████▀            #\n" \
+		"#             ▐████████▀▀████▀▀▌  █▌             #\n" \
+		"#            ▄██████  █  ▐███▄▄█▄▄██             #\n" \
+		"#            ████████▀▀▀█▀████▀    ██            #\n" \
+		"#            █████████    ▐███     ██            #\n" \
+		"#              ▀███▄█▄▄▄▄▄███▀▀▀▀▀▀▀             #\n" \
+		"#                                                #\n" \
+		"# Oops, something bad must have happened.        #\n" \
+		"# Starting busybox rescue and recovery shell.    #\n" \
+		"# Tip: type help<enter> for available commands.  #\n" \
+		"#                                                #\n" \
+		"##################################################\n"
+	${BB_BIN} sh
+}
+
+critical_error()
+{
+	echo "A critical error has occurred, entering recovery mode"
+	rescue_shell
+	shutdown
+}
+
+boot_root()
+{
+	echo "Mounting ${root}."
+	mount -t ${rootfstype} -o exec,suid,dev,noatime,$rootflags,$ro "${root}" "${ROOT_MOUNT}"
+	kernel_umount
+	echo "Starting linux on ${root} of type ${rootfstype} with init=${init}."
+	exec switch_root /mnt/root "${init}"
+}
+
+find_and_run_update()
+{
+	echo "Checking for updates ..."
+	for dev in ${UPDATE_DEVICES}; do
+		if [ ! -b "${dev}" ]; then
+			continue
+		fi
+
+		echo "Attempting to mount ${dev}."
+		if ! mount -t f2fs,ext4,vfat,auto -o exec,noatime "${dev}" "${UPDATE_MOUNT}"; then
+			continue
+		fi
+		
+		if [ ! -x "${UPDATE_SCRIPT}" ]; then
+			umount "${dev}"
+			echo "No executable update '${UPDATE_SCRIPT}' found on ${dev}, trying next."
+			continue
+		fi
+		
+		echo "Found update on ${dev}, executing update ${UPDATE_SCRIPT}."
```

Comment:
show 'starting update splash' though we very much probably prefer to let that up to the update script to show it's own splash screen